### PR TITLE
buf: do not expand $() in buf_expand_shell_variables()

### PR DESCRIPTION
**Pull request by johanmalm**

### CODE REVIEW

**Comment by Consolatis** [comment]
LGTM, tested and seems to work fine.

File: src/common/buf.c [hunk 1]
```diff
@@ -15,6 +16,39 @@ strip_curly_braces(char *s)
...
+	buf_init(&new);
+	for (int i = 0 ; i < s->len ; i++) {
+		if (s->buf[i] == '~') {
+			buf_add(&new, getenv("HOME"));
+		} else {
+			buf_add_one_char(&new, s->buf[i]);
+		}
+	}
+	free(s->buf);
+	s->buf = new.buf;
```
Comment (Consolatis) [inline src/common/buf.c:43]: Should we also copy over new.len` here?

**Comment by Consolatis** [comment]
> This increases safety

I don't really see this increasing safety.

File: src/common/buf.c (hunk 1 above)
Comment (johanmalm) [reply src/common/buf.c:31]: Great catch. Done.

**Comment by johanmalm** [comment]
> > This increases safety
> 
> I don't really see this increasing safety.

Commit message amended to remove the safety bit.

File: src/common/buf.c (hunk 1 above)
Comment (johanmalm) [reply src/common/buf.c:31]: Wrote a unit test to build confidence around this.
Realized that we need to do the same (copy `new.len` + `new.alloc`) for `buf_expand_shell_variables()`.
It's obvious in hindsight and I don't know why I didn't see it. Amazing what a unit test can do :smile:

**Comment by Consolatis** [comment]
LGTM, not tested.

**Pull request merged**